game-core/src/main/java/games/strategy/triplea/delegate/battle/end/NoMoreUnits.java

```diff
@@ -0,0 +1,77 @@
+package games.strategy.triplea.delegate.battle.end;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Properties;
+import games.strategy.triplea.delegate.Matches;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Predicate;
+import lombok.Builder;
+import lombok.NonNull;
+import org.triplea.java.collections.CollectionUtils;
+
+@Builder
+public class NoMoreUnits {
+
+  public enum Winner {
+    ATTACKER,
+    DEFENDER,
+    DEFENDER_TURN_1,
+    UNDETERMINED
```

Comment:
`UNDETERMINED` was not super obvious to me at first. I wondered if perhaps it meant an error happened, or if the battle could not determine a winner because the battle is "weird" (like air vs sub that cannot be hit). Would `NONE` maybe be a more clear name here with the same meaning? If not, are there other cases where `UNDETERMINED` is still a better fit?

Reply:
`NONE` is fine.  Just needed a flag to indicate that the winner isn't known yet.